File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        if(numRows == 1) return {{1}};
        vector<vector<int>>answer = {{1}, {1,1}};
        for(int i = 3 ; i<=numRows ; ++i){
            vector<int>v;
            v.push_back(1);
            for(int j = 0 ; j < answer[i-2].size()-1 ; ++j){
                v.push_back(answer[i-2][j] + answer[i-2][j+1]);
            }
            v.push_back(1);
            answer.push_back(v);
        }
        return answer;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        IList<IList<int>> ans = new List<IList<int>>();
        ans.Add(new List<int>{1});
        if(numRows == 1) return ans;
        ans.Add(new List<int>{1,1});
        for(int i = 2 ; i < numRows ; ++i){
            List<int>temp = new List<int>{1};
            for(int j = 0 ; j < ans[i-1].Count-1 ; ++j){
                temp.Add(ans[i-1][j] + ans[i-1][j+1]);
            }
            temp.Add(ans[i-1][ans[i-1].Count-1]);
            ans.Add(temp);
        }
        return ans;
    }
}